construction.tex

```diff
@@ -1,8 +1,13 @@
 \section{Construction}\label{section:construction}
 
+We implement the blockchain address protocol for Bitcoin P2PKH transactions. The implementation is parameterized by a secure signature scheme $S$, and a hash function $H$.
+$\GenAddress$ uses $S$ to generate a keypair, and hashes the public key in order to generate the public key hash. A tuple consisting of the public key hash and the secret key is returned.
+$\SpendVerify$ takes an outpoint $m$, a signature $\sigma$ and a public key hash $pkh$. The signature should contain the public key $pk$ corresponding to $pkh$ such that $H(pk) = pkh$ and a valid signature for the outpoint $m$, $\sigma'$. If these conditions are met the function returns true, otherwise it returns false.
```

Comment:
$m$ is not an output – it is the new transaction to which the money is being sent